dpbench/infrastructure/benchmark_runner.py

```diff
@@ -272,27 +266,21 @@ def run_benchmark(
         if results.error_state != ErrorCodes.SUCCESS:
             return (results, {})
 
-        output = {}
+        output = results_dict.get("outputs", {})
         results.setup_time = results_dict["setup_time"]
         results.warmup_time = results_dict["warmup_time"]
         results.exec_times = results_dict["exec_times"]
-
-        if "outputs" in results_dict:
-            output_npz = results_dict["outputs"]
-            if output_npz:
-                with np.load(output_npz) as npzfile:
-                    for outarr in npzfile.files:
-                        output.update({outarr: npzfile[outarr]})
-                os.remove(output_npz)
-            results.teardown_time = results_dict["teardown_time"]
-
-        if "return-value" in results_dict:
-            output.update({"return-value": results_dict["return-value"]})
+        results.teardown_time = results_dict["teardown_time"]
 
         if rc.validate and results.error_state == ErrorCodes.SUCCESS:
-            if bench._validate_results(
-                rc.preset, framework, output, rc.precision
-            ):
+            ref_framework = build_framework(rc.ref_framework)
+            ref_output = _exec_simple(
+                bench,
+                ref_framework,
+                rc.benchmark.reference_implementation_postfix,
+                rc.preset,
+            )
+            if validate_results(ref_output, output):
```

Comment:
design-wise the only problem when we can have is if we want to use a custom validation function for a specific benchmark.

Reply:
Related #156. We can merge the PR and address #156 later.